pkg/monitor/controller_test.go

```diff
@@ -0,0 +1,624 @@
+			defer close(stopCh)
+			err := controller.controllerMonitor.syncNode(tt.key)
+			if tt.expectedError != "" {
+				assert.Equal(t, tt.expectedError, err.Error())
+			} else {
+				require.NoError(t, err)
+				if tt.expectedAgentCRD == nil {
+					_, name, _ := splitKeyFunc(tt.key)
+					_, err := controller.controllerMonitor.client.CrdV1beta1().AntreaAgentInfos().Get(ctx, name, metav1.GetOptions{})
+					assert.Equal(t, true, errortesting.IsNotFound(err))
+				} else {
+					crd, err := controller.controllerMonitor.client.CrdV1beta1().AntreaAgentInfos().Get(ctx, tt.expectedAgentCRD.Name, metav1.GetOptions{})
+					require.NoError(t, err)
+					assert.Equal(t, tt.expectedAgentCRD, crd)
+				}
+			}
+		})
+	}
+}
+
+func TestDeleteAgentCRD(t *testing.T) {
+	ctx := context.Background()
+	testAgentCRD := &v1beta1.AntreaAgentInfo{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            "testAgentCRD",
+			ResourceVersion: "0",
+			Generation:      0,
+		},
+		NetworkPolicyControllerInfo: v1beta1.NetworkPolicyControllerInfo{
+			NetworkPolicyNum:  1,
+			AddressGroupNum:   1,
+			AppliedToGroupNum: 1,
+		},
+	}
+	tc := []struct {
+		name             string
+		existingAgentCRD *v1beta1.AntreaAgentInfo
+		prepareReactor   func(clientset *fakeclientset.Clientset)
+		expectedError    string
+	}{
+		{
+			name:             "CRD AntreaAgentInfo with given name exists",
+			prepareReactor:   func(clientset *fakeclientset.Clientset) {},
+			existingAgentCRD: testAgentCRD,
+			expectedError:    "",
+		},
+		{
+			name:           "no-ops when the CRD AntreaAgentInfo with given name does not exist",
+			prepareReactor: func(clientset *fakeclientset.Clientset) {},
+			expectedError:  "",
+		},
+		{
+			name: "fail to delete CRD AntreaAgentInfo with given name due to error",
+			prepareReactor: func(clientset *fakeclientset.Clientset) {
+				clientset.PrependReactor("delete", "antreaagentinfos", func(action cgtesting.Action) (handled bool, ret runtime.Object, err error) {
+					return true, &v1beta1.AntreaAgentInfo{}, errors.New("error deleting agent crd")
+				})
+			},
+			expectedError: "error deleting agent crd",
+		},
+	}
+	for _, tt := range tc {
+		t.Run(tt.name, func(t *testing.T) {
+			clientset := fakeclientset.NewSimpleClientset()
+			if tt.existingAgentCRD != nil {
+				clientset = fakeclientset.NewSimpleClientset(tt.existingAgentCRD)
+			}
+			tt.prepareReactor(clientset)
+			controller := newControllerMonitor(clientset)
+			stopCh := make(chan struct{})
+			initController(controller, stopCh)
+			defer close(stopCh)
+			err := controller.controllerMonitor.deleteAgentCRD("testAgentCRD")
+			if tt.expectedError != "" {
+				assert.Equal(t, tt.expectedError, err.Error())
+			} else {
+				require.NoError(t, err)
+				_, err = controller.controllerMonitor.client.CrdV1beta1().AntreaAgentInfos().Get(ctx, "testAgentCRD", metav1.GetOptions{})
+				assert.Equal(t, "antreaagentinfos.crd.antrea.io \"testAgentCRD\" not found", err.Error())
```

Comment:
you can use `assert.EqualError`

actually, can't you use `errortesting.IsNotFound` here as well?